visualization/SensorVisualization/RoboSailSerialParser.pde

```diff
@@ -0,0 +1,87 @@
+/*
+ * RoboSailSerialParser
+ *
+ * (c) 2016 Brian Dolan, RoboSail
+ *
+ * (See description for base class, SerialParser.)  The RobSailSerialParser implementation works for 
+ * the "real" output from one of the standard RoboSail Arduino programs used in the RoboSail program.
+ */
+import processing.serial.*;
+
+public class RoboSailSerialParser extends SerialParser
+{
+  RoboSailSerialParser(PApplet parent, String comPort)
+  {
+    super(parent, comPort);
+  }
+  
+  // parse the serial output, in the format of the Arduino program's output, which is
+  // int the format:
+  // 
+  // [joystick x value (0-1023)] [joystick y value (0-1023)] [B1][B2][B3][B4][B5]
+  //
+  // where B1-B1 are 0 or 1 values representing the state of the 5 buttons 
+  // 
+  // this function must be "hooked up" to the global serialEvent() function by the main 
+  // application, e.g. 
+  // void serialEvent(Serial sPort) { serialParser.serialEvent(sPort); }
```

Comment:
I was thinking that for simplicity, this program could parse the human-readable serial output from BoatCodeStarter[...].ino - the idea was that the students could use this program to help debug/write code if they want, or they could just look at the text output itself from the Arduino, without having to have some switch in the code to control the type of serial output (or requiring a second serial port). What do you think? (This is also the reason I had the baud rate set to 9600 - it is set that way in BoatCodeStarter[...].ino.)